src/linear_algebra/quotient.lean

```diff
@@ -123,6 +124,47 @@ begin
   simpa using not_mem_s
 end
 
+/-- More general, but harder to infer, instance than `submodule.quotient.module`. -/
+instance submodule.quotient.module' (R : Type*) {A : Type*} [ring A] [comm_ring R]
+  [algebra R A] [module A M] (P : submodule A M) :
+  module R P.quotient :=
+module.of_core
+{ smul := λ c x, algebra_map R A c • x, -- Note that this has to be defeq to `c • x` if `A = R`
```

Comment:
This should be using `is_scalar_tower` not algebra_map, and taking in a `[has_scalar R M` argument.

Reply:
Ah, that's indeed quite a bit nicer. I think we need `mul_action`, not `has_scalar`, to make sure everything is coherent.